### ISSUE TITLE
[Bug] LinearGradientBrush: StartPoint/EndPoint do nothing on a BoxView on Android

### ISSUE THREAD

**Issue by Tommigun1980** [xamarin/Xamarin.Forms#13493]
### Description

LinearGradientBrush's StartPoint and EndPoint properties have no effect on Android (while they do work properly on iOS).

### Steps to Reproduce

Define a brush such as:
```xaml
	<LinearGradientBrush
		x:Key="SomeBrush"
		StartPoint="1.8, 0" EndPoint="0, 1">
		<GradientStop Color="Red" Offset="0.0" />
		<GradientStop Color="Green" Offset="1.0" />
	</LinearGradientBrush>
```

Draw with it, for example:
```xaml
                <BoxView
                    Background="{StaticResource SomeBrush}"
                    HorizontalOptions="FillAndExpand"
                    VerticalOptions="FillAndExpand" />
```

_Edit: Issue happens only with a 'BoxView'; 'Image' for example works properly._

iOS - correct:
<img width="560" alt="Screenshot 2021-01-21 at 19 35 32" src="https://user-images.githubusercontent.com/60286968/105388888-dad3fe80-5c1f-11eb-8e92-35ad02a340ec.png">

Android - incorrect (StartPoint and EndPoint have no effect):
<img width="722" alt="Screenshot 2021-01-21 at 19 36 03" src="https://user-images.githubusercontent.com/60286968/105388947-ec1d0b00-5c1f-11eb-9cbe-c96f83220b9b.png">

### Expected Behavior

Android should honor StartPoint and EndPoint, as is done on iOS.

### Actual Behavior

StartPoint and EndPoint are ignored on Android.

### Basic Information

- Version with issue: Xamarin.Forms 5.0.0-pre4.
- Last known good version: None.
- Platform Target Frameworks: 
  - Android:  Android 11

### Environment

<details><summary>Show/Hide Visual Studio info</summary>

```
Visual Studio Community 2019 for Mac (Preview)
Version 8.9 Preview (8.9 build 1451)
Installation UUID: 63b853bd-57ca-4209-927d-80c6408f402c
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 6.18.0.23 (d16-6 / 088c73638)

	Package version: 612000113

Mono Framework MDK
Runtime:
	Mono 6.12.0.113 (2020-02/4fdfb5b1fd5) (64-bit)
	Package version: 612000113

Xamarin Designer
Version: 16.9.0.266
Hash: c4842c761
Branch: remotes/origin/c4842c761b9b6a95407f72278ca7fb42f8f7fdf0
Build date: 2021-01-07 06:17:08 UTC

Roslyn (Language Service)
3.9.0-3.20619.14+df59a33fd9beff9790e01a2a1ab21e4a1e6921b3

NuGet
Version: 5.8.0.6860

.NET Core SDK
SDK: /usr/local/share/dotnet/sdk/5.0.102/Sdks
SDK Versions:
	5.0.102
	5.0.101
	5.0.100
	3.1.405
	3.1.404
	3.1.403
MSBuild SDKs: /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Sdks

.NET Core Runtime
Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	5.0.2
	5.0.1
	5.0.0
	3.1.11
	3.1.10
	3.1.9

.NET Core 3.1 SDK
SDK: 3.1.405

Xamarin.Profiler
Version: 1.6.15.68
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

Updater
Version: 11

Apple Developer Tools
Xcode 12.2 (17535)
Build 12B45b

Xamarin.Mac
Xamarin.Mac not installed. Can't find /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/Version.

Xamarin.iOS
Version: 14.9.0.27 (Visual Studio Community)
Hash: f4c9327fa
Branch: main
Build date: 2020-11-19 10:57:31-0500

Xamarin.Android
Version: 11.2.0.0 (Visual Studio Community)
Commit: xamarin-android/d16-9/f908d16
Android SDK: /Users/tommikiviniemi/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		None installed

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 30.0.4
SDK Build Tools Version: 30.0.2

Build Information: 
Mono: 5e9cb6d
Java.Interop: xamarin/java.interop/d16-9@1d382be
ProGuard: Guardsquare/proguard/v7.0.1@912d149
SQLite: xamarin/sqlite/3.32.2@cfe06e0
Xamarin.Android Tools: xamarin/xamarin-android-tools/main@ad80a42

Microsoft OpenJDK for Mobile
Java SDK: /Users/tommikiviniemi/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_1.8.0.25
1.8.0-25
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

Android SDK Manager
Version: 16.9.0.21
Hash: 57e40ba
Branch: remotes/origin/main
Build date: 2021-01-08 01:57:14 UTC

Android Device Manager
Version: 16.9.0.14
Hash: 0fdccda
Branch: remotes/origin/main
Build date: 2021-01-08 01:57:36 UTC

Build Information
Release ID: 809001451
Git revision: cfd15313a6388ef8dada0182e22a058131c46f9d
Build date: 2021-01-15 08:42:21-05
Build branch: release-8.9
Xamarin extensions: cfd15313a6388ef8dada0182e22a058131c46f9d

Operating System
Mac OS X 10.16.0
Darwin 20.2.0 Darwin Kernel Version 20.2.0
    Wed Dec  2 20:39:59 PST 2020
    root:xnu-7195.60.75~1/RELEASE_X86_64 x86_64
```

</details>

**Comment by jsuarezruiz** [comment]
@Tommigun1980 Happens with `BoxView` or other specific View?

**Comment by Tommigun1980** [comment]
> @Tommigun1980 Happens with `BoxView` or other specific View?

Hi @jsuarezruiz! I had seen the issue with a BoxView, but encouraged by your question replaced the BoxView with an Image and lo and behold that works, so apparently this is an issue with BoxViews only.

**Comment by jsuarezruiz** [comment]
Thanks for the detailed feedback @Tommigun1980. The issue will be fixed by https://github.com/xamarin/Xamarin.Forms/pull/11812/

**Comment by Tommigun1980** [comment]
> Thanks for the detailed feedback @Tommigun1980. The issue will be fixed by #11812

Ah thank you! And apparently I had already reported this exact issue earlier, couldn't remember lol.

**Comment by jsuarezruiz** [comment]
They were not exactly the same, similar ones although the fix is the same. In any case, I hope we can include the fix soon!.
